methods/plot_results_in_box_plot.py

```diff
@@ -21,44 +21,50 @@
         for chosen_metric in __chosen_metrics:
             figure_method = plot_boxplot_by_method(data_to_plot, chosen_metric)
             figure_city = plot_boxplot_by_city(data_to_plot, chosen_metric)
-            stock_company_name = data_to_plot["subset_row"][0]
+            plot_params = get_plot_params(
+                figure_method, chosen_metric, data_to_plot, dataset_name
+            )
+            plot_params = get_plot_params(
+                figure_city, chosen_metric, data_to_plot, dataset_name
+            )
             print(f"Saving {dataset_name} {chosen_metric} boxplot plot")
-            input_map = {
-                ("Stock price", "KO"): (
-                    figure_method,
-                    "Stock price",
-                    "by_method_value",
-                    chosen_metric,
-                    "by_data_value",
-                ),
-                ("Stock price", "AAPL"): (
-                    figure_method,
-                    "Stock price",
-                    "by_method_growth",
-                    chosen_metric,
-                    "by_data_growth",
-                ),
-                ("India city pollution", stock_company_name): (
-                    figure_method,
-                    "Indian city pollution",
-                    "by_method",
-                    chosen_metric,
-                    "by_data",
-                ),
-            }
-            input_key = (dataset_name, stock_company_name)
-            if input_key in input_map:
-                figure_params = input_map[input_key]
-                save_plot_boxplot(*figure_params[:-1]),
-                save_plot_boxplot(
-                    figure_city, figure_params[1], figure_params[4], chosen_metric
-                ),
+            # input_map = {
+            #     ("Stock price", "KO"): (
+            #         figure_method,
+            #         "Stock price",
+            #         "by_method_value",
+            #         chosen_metric,
+            #         "by_data_value",
+            #     ),
+            #     ("Stock price", "AAPL"): (
+            #         figure_method,
+            #         "Stock price",
+            #         "by_method_growth",
+            #         chosen_metric,
+            #         "by_data_growth",
+            #     ),
+            #     ("India city pollution", stock_company_name): (
+            #         figure_method,
+            #         "Indian city pollution",
+            #         "by_method",
+            #         chosen_metric,
+            #         "by_data",
+            #     ),
+            # }
+            # input_key = (dataset_name, stock_company_name)
+            # if input_key in input_map:
+            #     figure_params = input_map[input_key]
+            save_plot_boxplot(*plot_params[:-1]),
+            save_plot_boxplot(
+                figure_city, plot_params[1], plot_params[4], chosen_metric
+            ),
```

Comment:
Can you add comments to explain what is going on here? Why plotting twice to boxplot?